### PR TITLE
526 v2 downtime notifications

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
Looks like this is already built in!

## Testing done
None. I'm not sure how to test this, actually. Thoughts?

## Screenshots
:grimacing:

## Acceptance criteria
- [x] The appropriate services are connected to the app

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by cvalarida** [comment]
It looks like this renders the downtime notification instead of two things when a service is unavailable:
1) The intro page
2) The submit button

@jbalboni Is that accurate?

**Comment by jbalboni** [comment]
@cvalarida That's correct, though hopefully the recent intro page changes haven't messed anything up there. 

For testing, we've generally I think modified the maintenance_windows controller in vets-api to return some info, but I can't find any docs around that. Maybe @ncksllvn remembers if that exists?

**Comment by ncksllvn** [comment]
Yeah I would just modify the [controller](https://github.com/department-of-veterans-affairs/vets-api/blob/ce59a2df1ad2e3ec6cbecad42ac3709e12d79ac7/app/controllers/v0/maintenance_windows_controller.rb#L7) to return [static JSON](https://github.com/department-of-veterans-affairs/vets-api/blob/ce59a2df1ad2e3ec6cbecad42ac3709e12d79ac7/spec/support/pagerduty/maintenance_windows_page_1.json#L1) to test. Probably the hackiest way of doing it 😆

**Comment by cvalarida** [comment]
Awesome! Thanks @jbalboni @ncksllvn!

I'll do that and get some screenshots in here.

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -118,6 +120,9 @@ const formConfig = {
   }/v0/disability_compensation_form/submit_all_claim`,
   submit: submitFormFor('disability-526EZ'),
   trackingPrefix: 'disability-526EZ-',
+  downtime: {
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/config/form.js]: Well that's convenient 😛

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ import environment from '../../../../platform/utilities/environment';
 import FormFooter from '../../../../platform/forms/components/FormFooter';
 import preSubmitInfo from '../../../../platform/forms/preSubmitInfo';
 
+import services from 'platform/monitoring/DowntimeNotification/config/externalServices';
```
Comment (markgreenburg) [inline src/applications/disability-benefits/all-claims/config/form.js]: Looks like you can just import `externalServices` from `platform/monitoring/DowntimeNotification`

**Review by markgreenburg — APPROVED** [review]
I think one of the import paths can be cleaned up but LGTM!

**Comment by cvalarida** [comment]
Hmm...It doesn't look like this actually blocks the start button. In my user profile, I've got this:
```js
{
  services: [
    'facilities',
    'hca',
    'edu-benefits',
    'form-save-in-progress',
    'form-prefill',
    'evss-claims',
    'user-profile',
    'appeals-status',
    'identity-proofed',
    'all_claims'
  ]
}
```
I'm missing several of the services I put into the `downtime.dependencies` config option, but I still see the start button.

Looking at [this line](https://github.com/department-of-veterans-affairs/vets-website/blob/526-v2-downtime-notifications-cv/src/platform/forms/save-in-progress/SaveInProgressIntro.jsx#L285), I see the `DowntimeNotification` wrapper is only used if the user isn't logged in. This is the opposite of what we're looking for.

@jbalboni Any idea why this is? It's been this way for a long time, it looks like.

**Comment by jbalboni** [comment]
@cvalarida Yeah, I think the way we set it up initially was that we didn't want to stop you from using the form if you were logged in, because you could save your info and come back later. You might not have prefill info, but that wasn't a big deal for a lot of forms. If you weren't logged in and a service was down, you wouldn't be able to submit, so we wanted to keep people out of the form.

This doesn't work for the 526, so we may need to add some kind of option to completely disable it when services are down.

**Review by bshyong — APPROVED** [review]

**Comment by cvalarida** [comment]
The downtime notification wrapper now wraps the start button.
image

**Review by bshyong — APPROVED** [review]

**Pull request merged**